wfl/Profiles/FoodOrderProfile.cs

```diff
@@ -60,6 +44,51 @@ protected override void Configure(EntityConfiguration<FoodOrder> configuration)
                             o => rc => rc.GetByIdAsync(o.RestaurantId)
                         )
                     )
+                )
+                .When(
```

Comment:
Okay, perhaps I missed where those checks were taking place.  What I actually think would make the most sense for our current use case is to handle that validation with a method on the business object.  Then the same business object can be used in the controller and the Hypermedia profile and all logic would be consolidated.  Even if it's something as simple as `CanRemove()`.